aten/src/ATen/native/mps/operations/Repeat.mm

```diff
@@ -44,12 +44,21 @@ void set_apparent_shapes(NSMutableArray<NSNumber*> * input_shape,
                          int64_t num_repeat_dims) {
 
 
-  // Set repeats_shape
+  bool repeat_empty = false;
+  if(num_repeat_dims == 0) {
+    num_repeat_dims = num_input_dims;
+    repeat_empty = true;
+  }
 
+  // Set repeats_shape
   repeats_shape = [NSMutableArray<NSNumber*> arrayWithCapacity:num_repeat_dims];
 
-  for(int i = 0; i < num_repeat_dims; i++)
-    repeats_shape[i] = [NSNumber numberWithInt:repeats[i]];
+  for(int i = 0; i < num_repeat_dims; i++) {
+    if(repeat_empty)
+      repeats_shape[i] = [NSNumber numberWithInt:1];
+    else
+      repeats_shape[i] = [NSNumber numberWithInt:repeats[i]];
```

Comment:
We should be using `[NSNumber numberWithInteger: .. `]. `numberWithInt` will truncate the number (`repeats[i]` is an int64_t) to an int32, and this would result to crashes for numbers larger than 2^31
Could you please add this change?